_playwright-tests/UI/helpers/helpers.ts

```diff
@@ -56,3 +60,50 @@ export const getRowCellByHeader = async (row: Locator, name: string) => {
 
   return row.getByRole('gridcell').nth(index);
 };
+
+export const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+export const waitForTaskPickup = async ({ request }: Page, repoUrl: string, type: string) => {
+  const response = await request.get(`/api/content-sources/v1/repositories/?url=${repoUrl}`);
+  expect(response.status()).toBe(200);
+  const body = await response.json();
+  expect(Array.isArray(body.data)).toBeTruthy();
+  const uuidList = body.data.map((data: { uuid: string }) => data.uuid) as string[];
+  expect(uuidList.length).toEqual(1);
+
+  const repoUuid = uuidList[0];
+  while (true) {
```

Comment:
If I am reading this correctly we could wait forever.... also the timeout on the request below is excessive, I'd expect that to return in 5s or so? 
Why would an api for tasks call take 60 seconds...

I like the idea of what you are doing here but lets have this while loop be based off off of a retry count, maybe configurable via passed in variable with a reasonable number set by default. maybe check this 10 times at maximum for example.